File: core/commands/registry/irules/psc__subscriber_id.py
# Enriched from F5 iRules reference documentation.
"""PSC::subscriber_id -- Get or set the subscriber id."""

from __future__ import annotations

from ....compiler.side_effects import ConnectionSide, SideEffect, SideEffectTarget
from .._base import CommandDef
from ..models import CommandSpec, FormKind, FormSpec, HoverSnippet, ValidationSpec
from ..namespace_models import EventRequires
from ..signatures import Arity
from ._base import _IRULES_ONLY, register

_SOURCE = "https://clouddocs.f5.com/api/irules/PSC-subscriber-id.html"


@register
class PscSubscriberIdCommand(CommandDef):
    name = "PSC::subscriber_id"

    @classmethod
    def spec(cls) -> CommandSpec:
        return CommandSpec(
            name="PSC::subscriber_id",
            dialects=_IRULES_ONLY,
            hover=HoverSnippet(
                summary="Get or set the subscriber id.",
                synopsis=("PSC::subscriber_id ((SUBSCRIBER_ID) (e164 |",),
                snippet="The PSC::subscriber_id command gets the subscriber id or sets the subscriber_id when the optional value is given.",
                source=_SOURCE,
                return_value="Return the subscriber id when no argument is given.",
            ),
            forms=(
                FormSpec(
                    kind=FormKind.DEFAULT,
                    synopsis="PSC::subscriber_id ((SUBSCRIBER_ID) (e164 |",
                ),
            ),
            validation=ValidationSpec(
                arity=Arity(),
            ),
            event_requires=EventRequires(),
            side_effect_hints=(
                SideEffect(
                    target=SideEffectTarget.NETWORK_IO,
                    reads=True,
                    connection_side=ConnectionSide.BOTH,
                ),
            ),
        )

File: rust/tcl-registry/src/commands/irules/psc__subscriber_id.rs
// tcl-lsp — a language server and toolchain for Tcl
// Copyright (C) 2026 James Deucker (bitwisecook) <https://github.com/bitwisecook>
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU Affero General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Affero General Public License for more details.
//
// You should have received a copy of the GNU Affero General Public License
// along with this program.  If not, see <https://www.gnu.org/licenses/>.
//
// SPDX-License-Identifier: AGPL-3.0-or-later

//! `PSC::subscriber_id` iRules command.
use crate::prelude::*;
pub const fn spec() -> CommandSpec {
    CommandSpec {
        name: "PSC::subscriber_id",
        dialects: Some(DialectSet::IRULES),
        arity: Arity::at_least(0),
        hover: Some(HoverSnippet {
            summary: "Get or set the subscriber id.",
            synopsis: &["PSC::subscriber_id ((SUBSCRIBER_ID) (e164 |"],
            snippet: "The PSC::subscriber_id command gets the subscriber id or sets the subscriber_id when the optional value is given.",
            source: "https://clouddocs.f5.com/api/irules/PSC-subscriber-id.html",
            examples: "",
            return_value: "Return the subscriber id when no argument is given.",
        }),
        forms: &[FormSpec {
            synopsis: "PSC::subscriber_id ((SUBSCRIBER_ID) (e164 |",
            ..FormSpec::DEFAULT
        }],
        side_effects: &[SideEffect {
            target: SideEffectTarget::NetworkIo,
            reads: true,
            connection_side: ConnectionSide::Both,
            ..SideEffect::DEFAULT
        }],
        ..CommandSpec::DEFAULT
    }
}
